File: server/src/main/scala/producer/message_generator/data_generators/RandomUuidV4Generator.scala
package producer.message_generator.data_generators

import com.tools.teal.pulsar.ui.producer.v1.producer as pb

case class RandomUuidV4Generator():
    def generate: String = java.util.UUID.randomUUID().toString

object RandomUuidV4Generator:
    def fromPb(v: pb.RandomUuidV4Generator): RandomUuidV4Generator =
        RandomUuidV4Generator()

    def toPb(v: RandomUuidV4Generator): pb.RandomUuidV4Generator =
        pb.RandomUuidV4Generator()

File: ui/components/ui/ProducerSession/producer-task/message-generator/data-generators/random-uuid-v4-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type RandomUuidV4Generator = {
  type: 'random-uuid-v4-generator'
};

export function randomUuidV4GeneratorFromPb(v: pb.RandomUuidV4Generator): RandomUuidV4Generator {
  return {
    type: 'random-uuid-v4-generator'
  };
}

export function randomUuidV4GeneratorToPb(v: RandomUuidV4Generator): pb.RandomUuidV4Generator {
  return new pb.RandomUuidV4Generator();
}
